pkg/node/node.go

```diff
@@ -25,9 +25,43 @@ type Node struct {
 
 // Lookup queries nodes looking for a particular node in the network
 func (n *Node) Lookup(ctx context.Context, to pb.Node, find pb.Node) ([]*pb.Node, error) {
+	c, err := n.getConnection(ctx, to)
+	if err != nil {
+		return nil, NodeClientErr.Wrap(err)
+	}
+
+	resp, err := c.Query(ctx, &pb.QueryRequest{Limit: 20, Sender: &n.self, Target: &find, Pingback: true})
+	if err != nil {
+		return nil, NodeClientErr.Wrap(err)
+	}
+
+	rt, err := n.dht.GetRoutingTable(ctx)
+	if err != nil {
+		return nil, NodeClientErr.Wrap(err)
+	}
+
+	if err := rt.ConnectionSuccess(&to); err != nil {
+		return nil, NodeClientErr.Wrap(err)
+
+	}
+
+	return resp.Response, nil
+}
+
+// Ping attempts to establish a connection with a node to verify it is alive
+func (n *Node) Ping(ctx context.Context, to pb.Node) (bool, error) {
+	_, err := n.getConnection(ctx, to)
```

Comment:
nothing is closing the connection or putting it back into the cache

Reply:
@jenlij has a PR coming up that will handle properly closing the connections to the pool. I'll make sure this connection is closed there.